### PR TITLE
Lint fixes and better validation for environment definition files

### PR DESCRIPTION
**Pull request by nbarrientos**

### CODE REVIEW

File: src/jens/environments.py [hunk 1]
```diff
@@ -164,50 +165,81 @@ def _create_new_environment(environment, inventory):
...
         environment = yaml.load(open(path, 'r'))
         for key in ('notifications',):
             if key not in environment:
-                raise JensEnvironmentsError("Missing '%s' in environemnt '%s'" %
+                raise JensEnvironmentsError("Missing '%s' in environment '%s'" %
                                             (key, environment))
-        if 'overrides' in environment and environment['overrides'] is None:
-                raise JensEnvironmentsError("Lacking overrides in environment '%s'" %
+
+        if 'default' in environment and type(environment['default']) != str:
```
Comment (davidmogar) [inline src/jens/environments.py]: Using isinstance here with six to support Python 2 and 3 is discarded right?

Comment (nbarrientos) [inline src/jens/environments.py]: Changed ;)

**Comment by nbarrientos** [comment]
Assigned to @jeromebelleman for a double check.

**Comment by jeromebelleman** [comment]
```
E:130,22: Instance of 'SyncManager' has no 'dict' member (no-member)            
E:131,21: Instance of 'SyncManager' has no 'Lock' member (no-member) 
```

in `jens.repos`. Probably depends on the Python version. (Not related to this particular merge request, mind you.)

**Comment by jeromebelleman** [comment]
OK, I've got nothing more to add.

File: src/jens/settings.py [hunk 2]
```diff
@@ -34,11 +34,12 @@ def parse_config(self, config_file_path):
...
             for error in flatten_errors(config, results):
-                section_list, key, msg = error
+                section_list, key, _ = error
                 section_string = '.'.join(section_list)
                 if key is not None:
-                    raise JensConfigError("Missing/not valid mandatory configuration key %s in section %s"
-                                          % (key, section_string))
+                    raise JensConfigError("Missing/not valid mandatory "
+                                          "configuration key %s in section %s",
+                                          key, section_string)
```
Comment (davidmogar) [inline src/jens/settings.py]: Well spotted. Kudos.

File: src/jens/environments.py [hunk 3]
```diff
@@ -379,21 +411,21 @@ def _annotate_environment(environment):
...
+    logging.debug("Removing cached hash for environment '%s'", environment)
     hash_cache_file = settings.CACHEDIR + "/environments/%s" % environment
     try:
         os.remove(hash_cache_file)
     # This shouldn't ever happen unless someone deleted the file or
     # changed its permissions externally
     except OSError, error:
-        logging.error("Couldn't remove cached hash for environemnt '%s'" %
-                      environment)
+        logging.error("Couldn't remove cached hash for environemnt '%s' (%s)",
```
Comment (nbarrientos) [inline src/jens/environments.py]: Fixed, thanks.

File: src/jens/settings.py (hunk 2 above)
Comment (nbarrientos) [inline src/jens/settings.py]: Nice one, fixed by reverting 5931d0d5172d07a9f17d16d831bd5ac717fabf14

File: bin/jens-gc [hunk 4]
```diff
@@ -23,21 +23,22 @@ def parse_cmdline_args():
     """Parses command line parameters."""
     parser = optparse.OptionParser()
     parser.add_option('-c', '--config',
-        help="Configuration file path (defaults to '/etc/jens/main.conf'",
-        default="/etc/jens/main.conf")
+                      help="Configuration file path "
+                           "(defaults to '/etc/jens/main.conf'",
```
Comment (nbarrientos) [inline bin/jens-gc]: It didn't know about that :) However, maybe better in a separate MR. Happy to review it :D

**Pull request merged**